src/main/java/core/basesyntax/services/strategy/impl/ReturnHandlerOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.services.strategy.impl;
+
+import core.basesyntax.db.DataStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.services.strategy.OperationHandler;
+
+public class ReturnHandlerOperation implements OperationHandler {
+    @Override
+    public void operate(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int quantityFruits = fruitTransaction.getQuantity();
+        if (DataStorage.fruitsStorageMap.containsKey(fruit)) {
+            int previousQuantity = DataStorage.fruitsStorageMap.get(fruit);
+            DataStorage.fruitsStorageMap.put(fruit, previousQuantity + quantityFruits);
+        } else {
+            DataStorage.fruitsStorageMap.put(fruit, quantityFruits);

```

Comment:
please read other Leonid comments carefully once more, he wrote about the `Map.merge()` method